### PR TITLE
Run clippy for jsonrpc_type

### PR DESCRIPTION
**Pull request by boundless-forest**

### CODE REVIEW

File: jsonrpc-types/src/rpc_types/specs/params.rs [hunk 1]
```diff
@@ -54,6 +54,14 @@ impl Params {
             Params::None => 0,
         }
     }
+
+    pub fn is_empty(&self) -> bool {
```
Comment (kaikai1024) [inline jsonrpc-types/src/rpc_types/specs/params.rs:58]: just to call `len`

**Review by kaikai1024 — CHANGES_REQUESTED** [review]

File: jsonrpc-types/src/rpc_types/specs/params.rs (hunk 1 above)
Comment (boundless-forest) [reply jsonrpc-types/src/rpc_types/specs/params.rs:58]: Done

File: jsonrpc-types/src/rpc_request/request.rs [hunk 2]
```diff
@@ -112,6 +109,7 @@ macro_rules! define_call {
 
         #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
         #[serde(untagged)]
+        #[allow(clippy::large_enum_variant)]   // TODO
```
Comment (kaikai1024) [inline jsonrpc-types/src/rpc_request/request.rs:112]: Need a real acceptable reason

Comment (boundless-forest) [reply jsonrpc-types/src/rpc_request/request.rs:112]: If we fix problem about `large_enum_variant` using `Box<>` in this pr . 
This pr must be unreadable.
Using Box<>, will influence all rpc reponse result, So, just use `#[allow]` temporary, i will fix it again in #224  after fix clippy issue.

File: jsonrpc-types/src/lib.rs [hunk 3]
```diff
@@ -34,6 +34,6 @@ mod macros;
 
 mod error;
 pub use error::{Error, ErrorCode};
-pub mod request;
-pub mod response;
-pub mod rpctypes;
+pub mod rpc_request;
+pub mod rpc_response;
+pub mod rpc_types;
```
Comment (boundless-forest) [inline jsonrpc-types/src/lib.rs:39]: Rename to maintain consistency, this need do adoption work in cita.

File: jsonrpc-types/src/rpc_request/request.rs (hunk 2 above)
Comment (kaikai1024) [reply jsonrpc-types/src/rpc_request/request.rs:112]: i'm not sure. TBD

you can continue to clippy the other crates.

File: jsonrpc-types/src/rpc_types/specs/params.rs [hunk 4]
```diff
@@ -54,6 +54,13 @@ impl Params {
...
```
Comment (kaikai1024) [inline jsonrpc-types/src/rpc_types/specs/params.rs:58]: Just `self.len() == 0`

**Comment by kaikai1024** [comment]
we'll keep this pr open, you can do other crates, then we'll disscuss the `allow` rule.

**Comment by boundless-forest** [comment]
> we'll keep this pr open, you can do other crates, then we'll disscuss the `allow` rule.

OK，  I keep eliminate clippy warning in other crate.

**Pull request closed without merging**